### PR TITLE
BBD-572: Should trim query in searches

### PR DESCRIPTION
**Pull request by zydc2010**

### CODE REVIEW

File: src/core/action-creator.ts [hunk 1]
```diff
@@ -121,8 +121,12 @@ export default class Creator {
 
   // request action creators
   updateSearch = (search: Actions.Search) =>
-    thunk<Actions.Search.UpdateSearch>(
-      Actions.UPDATE_SEARCH, search)
+    (dispatch: Dispatch<Actions.Search.UpdateSearch>) => {
+      const query = search.query ? search.query.trim() : '';
```
Comment (effervescentia) [inline src/core/action-creator.ts]: can be simplified to `const query = search.query && search.query.trim()`

File: test/unit/core/action-creator.ts [hunk 2]
```diff
@@ -327,12 +327,35 @@ suite('ActionCreator', ({ expect, spy, stub }) => {
...
-          const thunk = stub(utils, 'thunk');
+          const search: Actions.Search = { query: 'harry' };
+          const dispatch = spy();
+
+          actions.updateSearch(search)(dispatch);
+
+          expect(dispatch).to.be.calledWith({ type: Actions.UPDATE_SEARCH, ...<any>search });
+        });
+
+        it('should not create an UPDATE_SEARCH action when query string is empty', () => {
```
Comment (effervescentia) [inline test/unit/core/action-creator.ts]: I would combine this test with the last one and just dispatch multiple "invalid" actions to show that none of them trigger a dispatch

File: src/core/action-creator.ts (hunk 1 above)
Comment (jmbtutor) [inline src/core/action-creator.ts]: Updated

File: test/unit/core/action-creator.ts (hunk 2 above)
Comment (jmbtutor) [inline test/unit/core/action-creator.ts]: As previously discussed, we've decided against combining these because these test different expectations.

**Comment by jmbtutor** [comment]
Requested changes have been implemented.

**Pull request merged**